From 07e8ab7495e9f223ef3f192d60f45eec092f6fe4 Mon Sep 17 00:00:00 2001
From: Graham Ollis <plicease@cpan.org>
Date: Mon, 8 Sep 2014 12:09:06 -0400
Subject: [PATCH] use and propagate DESTDIR as appropriate

---
 lib/Alien/Base/ModuleBuild.pm | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/lib/Alien/Base/ModuleBuild.pm b/lib/Alien/Base/ModuleBuild.pm
index 5960502..a3b464b 100644
--- a/lib/Alien/Base/ModuleBuild.pm
+++ b/lib/Alien/Base/ModuleBuild.pm
@@ -323,8 +323,11 @@ sub ACTION_alien_install {
 
   return if $self->config_data( 'install_type' ) eq 'system';
 
+  my $destdir = $self->destdir;
+
   {
     my $target = $self->alien_library_destination;
+    $target = File::Spec->catdir($destdir, $target) if defined $destdir;
     local $CWD = $target;
 
     # The only form of introspection that exists is to see that the README file
@@ -337,13 +340,16 @@ sub ACTION_alien_install {
 
   {
     local $CWD = $self->config_data( 'working_directory' );
+    local $ENV{DESTDIR} = $destdir if defined $destdir;
     print "Installing library to $CWD ... ";
     $self->alien_do_commands('install') or die "Failed\n";
     print "Done\n";
   }
   
   if ( $self->alien_isolate_dynamic ) {
-    local $CWD = $self->alien_library_destination;
+    my $target = $self->alien_library_destination;
+    $target = File::Spec->catdir($destdir, $target) if defined $destdir;
+    local $CWD = $target;
     print "Isolating dynamic libraries ... ";
     mkdir 'dynamic' unless -d 'dynamic';
     foreach my $dir (qw( bin lib )) {
